File: src/main/java/me/miki/shindo/utils/file/filter/sound/Mp3FileFilter.java
package me.miki.shindo.utils.file.filter.sound;

import me.miki.shindo.utils.file.FileUtils;

import javax.swing.filechooser.FileFilter;
import java.io.File;

public class Mp3FileFilter extends FileFilter {

    @Override
    public boolean accept(File file) {

        if (file.isDirectory()) {
            return true;
        }

        String extension = FileUtils.getExtension(file);

        return extension != null && extension.equalsIgnoreCase("mp3");
    }

    @Override
    public String getDescription() {
        return "Sounds (*.mp3)";
    }
}

File: src/main/kotlin/com/shindoclient/shindo/utils/file/filter/sound/Mp3FileFilter.kt
package com.shindoclient.shindo.utils.file.filter.sound

import com.shindoclient.shindo.utils.file.FileUtils
import java.io.File
import javax.swing.filechooser.FileFilter

class Mp3FileFilter : FileFilter() {
    override fun accept(file: File?): Boolean {
        if (file == null) return false
        if (file.isDirectory) return true
        val extension = FileUtils.getExtension(file)
        return extension != null && extension.equals("mp3", ignoreCase = true)
    }

    override fun getDescription(): String = "MP3 Audio (*.mp3)"
}
